scripts/extension-version.js

```diff
@@ -43,13 +43,32 @@ function getDerivedExtensionVersion(version, isBeta = false) {
   )
 }
 
+function getExtensionVersionWithoutExtensionOnly(version, isBeta = false) {
```

Comment:
What does `getExtensionVersionWithoutExtensionOnly` mean?

Reply:
This is a helper function required to match the version. 

It converts `0.0.3.1` to `0.0.3`, so we can compare the extension version with a derived extension version. 

Reminder that derived version is the process of converting `2.0.0-beta.3` to `0.0.3`

Reply:
tl;dr it drops the extension only part from the VSCode extension

Reply:
I doubt the next person touching this code will be able to understand that from the function name, but 🤷

Reply:
I tried being verbose, happy to hear suggestions. I wanted to name it `getExtensionVersionWithoutExtensionOnlyPart` or something.

Reply:
Using "Extension" and "ExtensionOnly" in the same will always be confusing. Maybe find a better name for the latter? "ForthDigit" maybe? Does it have a name a la `major.minor.patch.???`?

Reply:
Semver spec calls it "Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format."

So, maybe `semVerWithoutMeta`?

Reply:
prerelease and meta are usually behind the `-`, so e.g. `-alpha.5+12ef9038` - I think that is something else than what we have here.

Reply:
True, what we have (x.y.z.a) is invalid semver. `node-semver` has a function for [coercion](https://github.com/npm/node-semver#coercion). 

I used `coerce` for actual coercion and called this function `coerceExtensionVersion`. 

Does that sound better?

Reply:
Yeah, the concept fits.